patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -48,6 +48,24 @@
  		public OverheadMessage chatOverhead;
  		public SelectionRadial DpadRadial = new SelectionRadial();
  		public SelectionRadial CircularRadial = new SelectionRadial(SelectionRadial.SelectionMode.RadialCircular);
+@@ -403,7 +_,16 @@
+ 		public int HotbarOffset;
+ 		public bool GoingDownWithGrapple;
+ 		public byte spelunkerTimer;
+-		public bool[] hideInfo = new bool[13];
++		private bool[] hideInfo = new bool[InfoDisplayLoader.InfoDisplayCount];
++		public bool[] HideInfo {
++			get {
++				if (hideInfo.Length != InfoDisplayLoader.InfoDisplayCount)
++					Array.Resize(ref hideInfo, InfoDisplayLoader.InfoDisplayCount);
```

Comment:
Can you elaborate on why this is necessary? `Player` instances shouldn't be live during mod reloading.

Reply:
There are a few other bool arrays in `Player` that get resized in this way (namely both incarnations of `adjTile`), so I thought it might be necessary for this as well. If it isn't, we can always just snap it out of existence.

Reply:
The adjTiles one might be an artifact of old bugs, we should see if we can get rid of this, this pattern is really annoying to code up.